### ISSUE TITLE
If a deployment fails before we can create anything on Connect, we still display an active `View Content` button

### ISSUE THREAD

**Issue by kgartland-rstudio** [posit-dev/publisher#2373]
If I try to deploy content to Connect but it fails before anything is created, under the failure message there's still a View Content button but it leads nowhere.

Here I was able to add the dogfood server using `http://` rather than `https://` and deploy but that deployment didn't result in content being created.  However, we still show the View Content button which doesn't lead anywhere.

![Image](https://github.com/user-attachments/assets/b01a108c-f872-46b1-a3f9-3a40a573d141)

**Comment by sagerb** [comment]
Based on a file refresh or manual refresh, we will be querying the status of content on the server. We can tie into this and disable the button if not possible.

**Comment by dotNomad** [comment]
Rather than relying on if the deployment record is in the pre-deployment state we can use the Connect API in a similar way to #2179

**Comment by dotNomad** [comment]
Handling this with a UI fix in #2522. The Go error itself has a separate issue: #2372
